retro/discrete_muon_kernels.py

```diff
@@ -118,3 +120,109 @@ def const_energy_loss_muon(hypo_params, limits=None, dt=1.0):
     pinfo_gen[:, 7] = dir_z * 0.562
 
     return pinfo_gen
+
+
+def table_energy_loss_muon(hypo_params, limits=None, dt=1.0):
+    """
+    Discrete-time track hypothesis that calculates dE/dx as the muon travels using a spline tabulated data
+
+    Use as a hypo_kernel with DiscreteHypo class.
+
+    Parameters
+    ----------
+    hypo_params : HypoParams*
+        Must have vertex (`.t`, `.x`, `.y`, and `.z), `.track_energy`,
+        `.track_azimuth`, and `.track_zenith` attributes.
+
+    limits
+        NOT IMPLEMENTED
+
+    dt : float
+        Time step in nanoseconds
+
+    Returns
+    -------
+    pinfo_gen : shape (N,8) numpy.ndarray, dtype float 32
+    """
+
+    track_energy = hypo_params.track_energy
+
+    #check for only cascade
+    if track_energy == 0:
+        pinfo_gen = np.array(
+            [[hypo_params.t,
+              hypo_params.x,
+              hypo_params.y,
+              hypo_params.z,
+              0,
+              0,
+              0,
+              0]],
+            dtype=np.float64
```

Comment:
Docstring says float32 ... and I use float64 elsewhere but I don't see why we need double precision here, so concert too float32 (unless you disagree that's sufficient)